libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -243,10 +243,36 @@ AP_Proximity::Status AP_Proximity::get_status(uint8_t instance) const
 
 AP_Proximity::Status AP_Proximity::get_status() const
 {
-    return get_status(primary_instance);
+    for (uint8_t i=0; i<num_instances; i++) {
+        const Status sensors_status = get_instance_status(i);
+        if (sensors_status != Status::Good) {
+            // return first bad status
+            return sensors_status;
+        }
+    }
+    // All valid sensors seem to be working
+    return Status::Good;
+}
+
+// prearm checks
+bool AP_Proximity::prearm_healthy(char *failure_msg, const uint8_t failure_msg_len) const
+{
+    for (uint8_t i=0; i<num_instances; i++) {
+        switch (get_instance_status(i)) {
+        case Status::NoData:
+            hal.util->snprintf(failure_msg, failure_msg_len, "PRX %X: No Data", i + 1);
```

Comment:
nitpick: %X is for uppercase hexadecimal numbers it seems.  Perhaps change this to "PRX%d:".  Save some flash by removing the space too.